pkg/controllers/user/cis/clusterScanHandler.go

```diff
@@ -75,35 +87,42 @@ func (csh *cisScanHandler) Create(cs *v3.ClusterScan) (runtime.Object, error) {
 			if cs.Spec.ScanConfig.CisScanConfig.DebugWorker {
 				appInfo.debugWorker = "true"
 			}
-			if cs.Spec.ScanConfig.CisScanConfig.Skip != "" {
+			if cs.Spec.ScanConfig.CisScanConfig.OverrideSkip != nil {
 				skipOverride = true
 			}
 		}
 
 		var cm *v1.ConfigMap
 		if skipOverride {
 			// create the cm
-			cm = &v1.ConfigMap{
-				TypeMeta: metav1.TypeMeta{
-					Kind: "ConfigMap",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name: getOverrideConfigMapName(cs),
-				},
-				Data: map[string]string{
-					ConfigFileName: cs.Spec.ScanConfig.CisScanConfig.Skip,
-				},
-			}
-			logrus.Infof("cisScanHandler: Create: creating skip override configmap %v with contents: %v", cm.Name, cs.Spec.ScanConfig.CisScanConfig.Skip)
-			for i := 0; i < NumberOfRetriesForConfigMapCreate; i++ {
-				cm, err = csh.configMapsClient.Create(cm)
-				if err == nil || kerrors.IsAlreadyExists(err) {
-					break
+			skipDataBytes, err := getOverrideSkipInfoData(cs.Spec.ScanConfig.CisScanConfig.OverrideSkip)
+			if err != nil {
+				logrus.Error("cisScanHandler: Create: error getting overrideSkip: %v", err)
+			} else {
+				cm = &v1.ConfigMap{
+					TypeMeta: metav1.TypeMeta{
+						Kind: "ConfigMap",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: getOverrideConfigMapName(cs),
+					},
+					Data: map[string]string{
+						ConfigFileName: string(skipDataBytes),
+					},
+				}
+				success := false
+				for i := 0; i < NumberOfRetriesForConfigMapCreate; i++ {
+					cm, err = csh.configMapsClient.Create(cm)
+					if err == nil || kerrors.IsAlreadyExists(err) {
+						logrus.Infof("cisScanHandler: Create: created skip override configmap %v with contents: %v", cm.Name, string(skipDataBytes))
+						success = true
+						break
+					}
+					time.Sleep(RetryIntervalInMilliseconds * time.Millisecond)
 				}
-				if err != nil {
-					logrus.Errorf("cisScanHandler: Create: (try: %v) error creating configmap: %v", i+1, err)
+				if !success {
+					logrus.Errorf("cisScanHandler: Create: error creating configmap: %v", err)
```

Comment:
If !success, do you also need to make cm instance nil so that it is not set here https://github.com/rancher/rancher/blob/393dadee4196a2ccbe31bde711df6a50f29a32ca/pkg/controllers/user/cis/clusterScanHandler.go#L135 ?

Reply:
I don't think it should be an issue coz if Create fails, it should set cm to nil, but anyways I can set it to nil explicitly.